src/arch/xtensa/include/arch/sof.h

```diff
@@ -28,6 +32,20 @@ static inline void *arch_get_stack_ptr(void)
 	return ptr;
 }
 
+static inline void *arch_get_stack_entry(void)
+{
+	volatile xtos_task_context *task_ctx = task_context_get();
```

Comment:
Does this need to be volatile ? Shouldn't task_context_get() handle any coherency ?

Reply:
It is switched by HAL, so probably better to set it volatile.